### PR TITLE
COR 340 PARTIAL AddRecordScreen Test

### PR DESCRIPTION
**Pull request by nilueps**
- setup
- setup
- add lint scripts
- setup tests
- split AddRecordScreen into screen and container
- update exports
- add record screen test

### CODE REVIEW

File: web/app/frontend/src/components/form/TextInput.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import {Text, TextInput as TextInputRNP} from "react-native-paper";
 import {darkTheme} from "../../constants/theme";
 import {InputProps} from "./FormControl";
 
-const TextInput: React.FC<InputProps> = (
+const TextInput = (
```
Comment (internetti) [inline web/app/frontend/src/components/form/TextInput.tsx:7]: why this change?

File: web/app/frontend/src/components/screens/AddRecordScreen.tsx [hunk 2]
```diff
@@ -1,106 +1,36 @@
...
+    form,
+    control,
+    onSubmit,
+    formState,
+    isWeb,
+    hasLocalData,
+    isConnected,
+    isLoading,
+}: AddRecordScreenProps)=> {
+  // console.log(form?.fields)
```
Comment (internetti) [inline web/app/frontend/src/components/screens/AddRecordScreen.tsx:30]:
```suggestion
```

File: web/app/frontend/src/components/form/FormControl.tsx [hunk 3]
```diff
@@ -8,51 +8,49 @@ import ReferenceInput from "./ReferenceInput";
...
-    value?: any,
-    control: Control<any, object>,
-    errors?: object
+    name: string;
+    fieldDefinition: FieldDefinition;
+    style?: ViewStyle;
+    value?: any;
+    control: Control<any, object>;
+    errors?: object;
+    testID?: string;
```
Comment (internetti) [inline web/app/frontend/src/components/form/FormControl.tsx:31]: since this is always "formControl", I don't think it needs to be passed as a parameter

File: web/app/frontend/src/components/screens/AddRecordScreen.tsx [hunk 4]
```diff
@@ -1,106 +1,36 @@
...
     return (
         <ScrollView contentContainerStyle={[layout.container, layout.body, common.darkBackground]}>
-
             <View style={[]}>
```
Comment (internetti) [inline web/app/frontend/src/components/screens/AddRecordScreen.tsx:33]:
```suggestion
            <View>
```

File: web/app/frontend/src/components/Router.tsx [hunk 5]
```diff
@@ -73,8 +74,8 @@ export default function Router() {
                                 title: routes.addRecord.title
                             }}
                         >
-                            {(props) =>
-                                <AddRecordScreen {...props} state={state} dispatch={dispatch}/>
+                            {({navigation, route}) =>
+                                <AddRecordScreenContainer navigation={navigation} route={route as AddRecordScreenContainerProps['route']} state={state} dispatch={dispatch}/>
```
Comment (internetti) [inline web/app/frontend/src/components/Router.tsx:78]: can the linting be set up so each property is on its own line?

File: web/app/frontend/__tests__/screens/AddRecordScreen.test.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    test("renders correctly", () => {
+        const { toJSON } = render(<WrappedComponent />);
+        expect(toJSON()).toMatchSnapshot();
+    });
+
+    test("displays all of a form's fields", () => {
+        const { getAllByTestId } = render(<WrappedComponent />);
+        expect(getAllByTestId(testIds.formControl).length).toEqual(mockForm.fields.length);
+    });
+});
```
Comment (internetti) [inline web/app/frontend/__tests__/screens/AddRecordScreen.test.tsx:57]: can you add tests for what is shown when hasLocalData, isConnected, isLoading are true/false?

**Review by internetti — CHANGES_REQUESTED** [review]

File: web/app/frontend/src/components/form/TextInput.tsx (hunk 1 above)
Comment (nilueps) [reply web/app/frontend/src/components/form/TextInput.tsx:7]: We're not sure, but somehow tsserver was complaining.

Also: https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components

File: web/app/frontend/src/components/form/FormControl.tsx (hunk 3 above)
Comment (nilueps) [reply web/app/frontend/src/components/form/FormControl.tsx:31]: I believe we may need it because it is being consumed by the Controller component within the FormControl stack

https://react-hook-form.com/api/useform/control

File: web/app/frontend/src/components/Router.tsx (hunk 5 above)
Comment (nilueps) [reply web/app/frontend/src/components/Router.tsx:78]: just tested it, it already does that. I just didn't run the fix on the whole package because it would make a massive PR. We should probably submit a dedicated PR for that.
I'll run it for the files in this PR though

File: web/app/frontend/src/components/form/FormControl.tsx (hunk 3 above)
Comment (internetti) [reply web/app/frontend/src/components/form/FormControl.tsx:31]: yes, it's needed, but it doesn't need to be a parameter because it doesn't change based on the field type. you can just put the constant in line 44 here.

Comment (nilueps) [reply web/app/frontend/src/components/form/FormControl.tsx:31]: I'm not sure I understand. Do you mean `const { control } = useForm()`?

Comment (internetti) [reply web/app/frontend/src/components/form/FormControl.tsx:31]: I mean you can just import the testIDs in this file instead of in the AddRecordScreen and use them in the line `<View style={{ margin: 10 }} testID={testIds.formControl}>`

**Comment by ludydoo** [comment]
@internetti 
Why ? Then all the snapshot tests will fail on the CICD pipeline...

**Comment by internetti** [comment]
> @internetti Why ? Then all the snapshot tests will fail on the CICD pipeline...

didn't think of that, nevermind then :+1:

**Review by internetti — DISMISSED** [review]

File: web/app/frontend/src/components/form/FormControl.tsx (hunk 3 above)
Comment (nilueps) [reply web/app/frontend/src/components/form/FormControl.tsx:31]: ah, yes you are correct, good catch

File: web/app/client/src/index.ts [hunk 7]
```diff
@@ -1,3 +1,5 @@
-export {default as Client} from "./lib/client"
-export * from "./lib/utils"
-export * from "./lib/types"
+import Client from './lib/client'
+
+export default Client
```
[the remark this answers is not in the document]
Comment (nilueps) [reply web/app/client/src/index.ts:3]: should be good, although I'm seeing some issues with the `core-auth` imports. might be better to fix this in a separate ticket though
